adb_graphics/figures/maps.py

```diff
@@ -394,6 +428,11 @@ def _draw_field(self, ax, field, func, **kwargs):
         x, y = self._xy_mesh(field)
         vals = field.values()
 
+        # Scatter plot dots are sized by value. Doing this here alters the size
+        # without altering the colors.
+        if self.plot_scatter:
+            vals = np.full_like(vals, np.log10(vals) * 20, dtype='float')
+
```

Comment:
It would be best to keep all this logic contained together with this special plot.

Reply:
I know and I've tried keeping this statement within the _draw_scatter function.  If I put the statement there, and comment out the one in _draw_field, I get this error:

  File "/mnt/lfs1/BMC/wrfruc/jamison/adb_graphics/adb_graphics/figures/maps.py", line 640, in _xy_mesh
    lat, lon = field.latlons()
AttributeError: 'numpy.ndarray' object has no attribute 'latlons'

Reply:
With my recent PR for handling diff plots, you should be able to set the `data` attribute of the field object. So, in `_draw_scatter` you can set the values you want to plot like this:

```
field.data = np.log10(field.values()) * 20
```

Then you should have the values you want to plot and the lat/lon attributes you need from the field object.

Reply:
Thank you for this, it worked!  Now all of the logic is contained within _draw_scatter.